### PR TITLE
[IRGen] Fix layout string generation for pre-specialized metadata

### PR DESCRIPTION
**Pull request by drexin**
rdar://108012057

Pre-specialized metadata has to be specifically handled by using the bound generic type instead of the unbound one. All the necessary information is already being passed down as BoundGenericTypeCharacteristics, we just need to apply them when present.

### CODE REVIEW

**Comment by drexin** [comment]
@swift-ci smoke test

**Comment by drexin** [comment]
@aschwaighofer I have so far been unable to reproduce the issue in a test.

**Review by aschwaighofer — APPROVED** [review]
LGTM

File: stdlib/public/runtime/Metadata.cpp [hunk 1]
```diff
@@ -2833,7 +2833,10 @@ void swift::swift_initStructMetadataWithLayoutString(
       previousFieldOffset = fieldType->vw_size();
       fullOffset += previousFieldOffset;
     } else if (fieldType->isAnyExistentialType()) {
-      auto tag = RefCountingKind::Existential;
+      auto *existential = dyn_cast<ExistentialTypeMetadata>(fieldType);
```
Comment (aschwaighofer) [inline stdlib/public/runtime/Metadata.cpp:2836]: 👍

**Comment by drexin** [comment]
@swift-ci smoke test

**Comment by drexin** [comment]
@swift-ci smoke test

**Pull request merged**